ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+        run_numbers = list(map(self._get_run_number, file_list))
+        self.run_numbers = np.unique(run_numbers)
+
+        # # Setting up the current run with the first run present in the data
+        # self.current_run = self._set_active_run(run_number=0)
+        self.current_run = None
+        
+        # MAGIC telescope positions in m wrt. to the center of CTA simulations
+        self.magic_tel_positions = {
+            1: [-27.24, -146.66, 50.00] * u.m,
+            2: [-96.44, -96.77, 51.00] * u.m
+        }
+        # MAGIC telescope description
+        optics = OpticsDescription.from_name('MAGIC')
+        geom = CameraGeometry.from_name('MAGICCam')
+        self.magic_tel_description = TelescopeDescription(optics=optics, camera=geom)
+        self.magic_tel_descriptions = {1: self.magic_tel_description, 2: self.magic_tel_description}
+        self.magic_subarray = SubarrayDescription('MAGIC', self.magic_tel_positions, self.magic_tel_descriptions)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return
+
+    @staticmethod
+    def is_compatible(file_mask):
+        """
+        This method checks if the specified file mask corresponds
+        to MAGIC data files. The result will be True only if all
+        the files are of ROOT format and contain an 'Events' tree.
+
+        Parameters
+        ----------
+        file_mask: str
+            A file mask to check
+
+        Returns
+        -------
+        bool:
+            True if the masked files are MAGIC data runs, False otherwise.
+
+        """
+
+        is_magic_root_file = True
+
+        file_list = glob.glob(file_mask)
+
+        for file_path in file_list:
+            try:
+                with uproot.open(file_path) as input_data:
+                    if 'Events' not in input_data:
+                        is_magic_root_file = False
+            except ValueError:
+                # uproot raises ValueError if the file is not a ROOT file
+                is_magic_root_file = False
+                pass
+
+        return is_magic_root_file
+
+    @staticmethod
+    def _get_run_number(file_name):
+        """
+        This internal method extracts the run number from
+        the specified file name.
+
+        Parameters
+        ----------
+        file_name: str
+            A file name to process.
+
+        Returns
+        -------
+        int:
+            A run number of the file.
+        """
+
+        mask = ".*\d+_M\d+_(\d+)\.\d+_.*"
+        parsed_info = re.findall(mask, file_name)
+
+        run_number = int(parsed_info[0])
+
+        return run_number
+
+    def _set_active_run(self, run_number):
+        """
+        This internal method sets the run that will be used for data loading.
+
+        Parameters
+        ----------
+        run_number: int
+            The run number to use.
+
+        Returns
+        -------
+
+        """
+
+        input_path = '/'.join(self.input_url.split('/')[:-1])
```

Comment:
I think this line should be `input_path = os.path.dirname(self.input_url)`

Reply:
I'm going to suggest at the framework meeting that we switch to pathlib for all file handling. It's so much nicer.
```py3
import pathlib

input_path = pathlib.Path(input_url)
input_path.exists()  # -> True
input_path.parent # -> the parent dirname
input_path.name # -> the filename 
input_path.stat().st_size # -> file size
```
etc.

Reply:
Awaiting for your coming discussion, I've implemented Max's proposal for now.